### PR TITLE
his qilib qo'yorasiz

### PR DESCRIPTION
**Pull request by tursunboev**

### CODE REVIEW

**Comment by tursunboev** [comment]
Yopildi

File: Product_API/Repositories/IProductPostgressRepository.cs [hunk 1]
```diff
@@ -1,17 +1,60 @@
...
         public Product Add(Product product)
         {
-            throw new NotImplementedException();
+            using (NpgsqlConnection connection = new NpgsqlConnection(ConnectionString))
+            {
+
+                connection.Open();
+                try
+                {
+                    using NpgsqlCommand cmd = new NpgsqlCommand(@$"insert into products(Name,Description,PhotoPath) values ('{product.Name}','{product.Description}','{product.PhotoPath}') ", connection);
```
Comment (Tohirjon-Odilov) [inline Product_API/Repositories/IProductPostgressRepository.cs:18]: Faqat shu table mavjud bo'lmasa ishkal bo'larkanda

File: Product_API/Repositories/IProductPostgressRepository.cs [hunk 2]
```diff
@@ -1,17 +1,60 @@
...
         }
 
         public List<Product> GetAll()
         {
-            throw new NotImplementedException();
+            List<Product> products = new List<Product>();
+            using (NpgsqlConnection connection = new NpgsqlConnection(ConnectionString))
+            {
+
+                connection.Open();
```
Comment (Tohirjon-Odilov) [inline Product_API/Repositories/IProductPostgressRepository.cs:37]: Shu baribir using ichida open qilib ochilishiga tushnmadim

**Review by Tohirjon-Odilov — APPROVED** [review]
Qoyil lekin

File: Product_API/Repositories/IProductPostgressRepository.cs [hunk 3]
```diff
@@ -1,17 +1,60 @@
-﻿using Product_API.Models;
+﻿using Npgsql;
+using Product_API.Models;
 
 namespace Product_API.Repositories
 {
     public class ProductPostgressRepository : IProductRepository
     {
+        public  string ConnectionString = "Host=localhost;Port = 5432;Database = Lesson;User Id = postgres;Password = root;";
```
Comment (dilshodbek-abdullayev) [inline Product_API/Repositories/IProductPostgressRepository.cs:8]: chota bo'mapti manga yoqmadi 😏

File: Product_API/Repositories/IProductPostgressRepository.cs [hunk 4]
```diff
@@ -1,17 +1,60 @@
...
-            throw new NotImplementedException();
+            List<Product> products = new List<Product>();
+            using (NpgsqlConnection connection = new NpgsqlConnection(ConnectionString))
+            {
+
+                connection.Open();
+                try
+                {
+                    Product product = new Product();
+                    using NpgsqlCommand cmd = new NpgsqlCommand(@$"select username from users ", connection);
```
Comment (Tohirjon-Odilov) [inline Product_API/Repositories/IProductPostgressRepository.cs:41]: usernama nima uchun kerak ekanligini ayta olasizmi va users table'ga necchi endi

**Review by Tohirjon-Odilov — CHANGES_REQUESTED** [review]

**Review by OzodbekOlimjonov1411 — APPROVED** [review]
Siiuuuuuuuuuuuuuuuu

**Pull request closed without merging**